### PR TITLE
[SPARK-13989][SQL] Remove non-vectorized/unsafe-row parquet record reader

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

This PR cleans up the new parquet record reader with the following changes:
1. Removes the non-vectorized parquet reader code from `UnsafeRowParquetRecordReader`.
2. Removes the non-vectorized column reader code from `ColumnReader`.
3. Renames `UnsafeRowParquetRecordReader` to `VectorizedParquetRecordReader` and `ColumnReader` to `VectorizedColumnReader`
4. Deprecate `PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED`
## How was this patch tested?

Refactoring only; Existing tests should reveal any problems.

### CODE REVIEW

**Comment by sameeragarwal** [comment]
cc @nongli

**Comment by nongli** [comment]
Can you undo the file rename? It makes this hard to review.

**Comment by sameeragarwal** [comment]
oops sorry, that was because I moved `ColumnReader` to a new file. Reverted that change!

**Comment by nongli** [comment]
Patch looks good. Can you look for places where we enable unsafe row and disable vectorized (maybe just grep files that have both flags) and remove what is not relevant? I think parquetreadbenchmark will have this for sure.

**Comment by sameeragarwal** [comment]
Thanks, all comments addressed!

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala [hunk 1]
```diff
@@ -51,8 +51,10 @@ private[sql] trait ParquetTest extends SQLTestUtils {
       (f: DataFrame => Unit) = {
     (true :: false :: Nil).foreach { vectorized =>
       if (!vectorized || testVectorized) {
-        withSQLConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED.key -> vectorized.toString) {
-          f(sqlContext.read.parquet(path.toString))
+        withSQLConf(SQLConf.PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED.key -> vectorized.toString) {
```
Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala]: Should we remove or deprecate PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED ?

Comment (sameeragarwal) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala]: yes, that'd be nice. Is it safe to remove or should I deprecate it?

Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala]: deprecate it if it's easy? Taking others as example.

Comment (sameeragarwal) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala]: Thanks, deprecated `PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED ` via b18ad0cbf3939edf8a4aed29bab5c4bb2f87de67

**Comment by davies** [comment]
LGTM, will merge it once it passed tests.

**Comment by davies** [comment]
Merging this into master, thanks!

**Pull request closed without merging**